src/test/libcephfs/test.cc

```diff
@@ -1317,6 +1319,80 @@ TEST(LibCephFS, Rename) {
   ceph_shutdown(cmount);
 }
 
+static void rename_racer_func(struct ceph_mount_info *cmount,
+                              std::condition_variable &cond,
+                              std::mutex &lock, bool &thread_started,
+                              char *src, char *dst)
+{
+  {
+    std::scoped_lock<std::mutex> l(lock);
+    thread_started = true;
+    cond.notify_one();
+  }
+
+  // the ceph_rename() will be stuck and will wait the
+  // client_inject_directory_dentries_race to be false
+  ASSERT_EQ(0, ceph_rename(cmount, src, dst));
+}
+
+TEST(LibCephFS, RenameRace) {
```

Comment:
cite the tracker ticket this is a reproducer for

Reply:
Fixed them all.